### DISCUSSION TITLE
Display images in a 2-column grid on mobile, just like on desktop

### DISCUSSION THREAD

**Discussion by hooziwang** [withastro/starlight#2018] [Q&A] [answered]
### What version of `starlight` are you using?

0.22.1

### What is your idea?

I put a lot of images in CardGrid, and I want the images to be displayed in 2 columns on the mobile side, just like on the PC side.
<img width="346" alt="截屏2024-06-15 19 37 03" src="https://github.com/withastro/starlight/assets/42474132/ce18a897-4343-42b1-b47a-82ba4c7ff19f">

how should I modify the code? Thank you

### Why is this feature necessary?

When there are a large number of images, two columns are displayed to facilitate user browsing.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment, marked as answer]
For something like this, it might be best to make a custom grid component that gives you full flexibility, for example:

```astro
---
// src/components/TwoColumnGrid.astro
---

<div class="my-grid">
  <slot />
</div>

<style>
.my-grid {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 1rem;
}
</style>
```

Then in your content you can use this:

```mdx

import TwoColumnGrid from '/src/components/TwoColumnGrid.astro';

<TwoColumnGrid>
[![](/1.webp)](/url_1/)
[![](/2.webp)](/url_2/)
[![](/3.webp)](/url_3/)
</TwoColumnGrid>
